### ISSUE TITLE
SIM800L: can not read RSSI on HASSIO with ESPHome

### ISSUE THREAD

**Issue by fahhome123** [esphome/issues#1753]
**Operating environment/Installation (Hass.io/Docker/pip/etc.):**

Hass.io

**ESP (ESP32/ESP8266, Board/Sonoff):**

ESP32 with SIM800L

**ESPHome version (latest production, beta, dev branch)**

ESPHome 1.15.3

**Affected component:**

**Description of problem:**
ESP32 with SIM800L can send and receive message. This works. 
I use it on HomeAssistant for security things.
I want to trigger the RSSI from SIM800L, but it not works.

RSSI is not readable.

**Problem-relevant YAML-configuration entries:**
```yaml
PASTE YAML FILE HERE

```

**Logs (if applicable):**

```
PASTE DEBUG LOG HERE

```

**Additional information and things you've tried:**

**Comment by glmnet** [comment]
Indeed, the rssi is read from the device but a sensor publishing the value is not available, can be done though.
Moving this to feature request.
